support/sier-codec/tests/integrated.rs

```diff
@@ -164,3 +164,55 @@ fn struct_() {
 
     assert_eq!(obj["foo"].as_u8(), Some(42));
 }
+
+const JSON_STRUCT_DEF: &'static str = r#"
+struct Foo {
+    bar :u64;
+    baz :string;
+    qux :List<u64>;
+    corge :Corge;
+}
+"#;
+
+const JSON_STRUCT_DEF2: &'static str = r#"
+struct Corge {
+    gz :u64;
+}
+"#;
+
+const JSON: &'static str = r#"
+{
+    "sierType": "Foo",
+    "bar": 42,
+    "baz": "abc",
+    "qux": [4, 2],
+    "corge": {
+        "sierType": "Corge",
```

Comment:
We shouldn't need a marker for the SIER type. We should be able to know the type from the struct defs.